File: 1441. Build an Array With Stack Operations/Build an Array With Stack Operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> result;
        int last = 0;
        int skip;
        for (int i = 0; i < target.size(); i++){
            skip = target[i] - last - 1;
            for (int j = 0; j < skip; j++){
                result.push_back("Push");
                result.push_back("Pop");
            }
            result.push_back("Push");
            last = target[i];
        }
        return result;
    }
};

File: 1441. Build an Array With Stack Operations/Build an Array With Stack Operations.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        result = []
        last = 0
        for i in target:
            skip = i - last - 1
            for j in range(skip):
                result.append("Push")
                result.append("Pop")
            result.append("Push")
            last = i
        return result
